core/Maybe.carp

```diff
@@ -55,9 +55,10 @@ It is the inverse of [`just?`](#just?).")
   (doc ptr "Creates a `(Ptr a)` from a `(Maybe a)`. If the `Maybe` was
 `Nothing`, this function will return a `NULL` value.")
   (defn ptr [a]
-    (match a
-      (Nothing) NULL
-      (Just x) (address x)))
+    (the (Ptr a) (Unsafe.coerce
+      (match @a
```

Comment:
There are plans to make matching work on refs very soon, this would use that – right?

Reply:
Can I read about that somewhere?

Reply:
https://github.com/carp-lang/Carp/issues/469

Reply:
Yeah, I guess the implementation would change then.